### PR TITLE
6188-add unit test for announcemen covering issuance from exception handler block

### PR DESCRIPTION
**Pull request by amiracam**
experimenting with contributing to Pharo by pushing a borrowed Sunit for Announcements

Note, could not figure how to get past the "INVALID-ISSSUE" title being assigned , obviously the lookup is failing for some reason

### CODE REVIEW

**Comment by guillep** [comment]
> Note, could not figure how to get past the "INVALID-ISSSUE" title being assigned , obviously the lookup is failing for some reason

Sorry about that. The issue title is fetched from the selected remote in the dropdown. There is an issue open here: https://github.com/pharo-vcs/iceberg/issues/1364. If you have ideas or proposals of how to make it more understandable, we are open to ideas.

Anyways, congrats on your first PR! :)

**Review by guillep — APPROVED** [review]
Thanks!

Fix #6188

**Comment by amiracam** [comment]
This is probably the proudest moment of my life !!!
Seriously, wonder what would be more beneficial adding a major trading platform to the ranks of happy Pharo users or me fixing some bugs with some menus ?

Not that I mind fixing bugs, but I'm swamped with my day gig, but I'll look at those easy issues.
BTW, what's the usual process would I delete my fork and just re-fork and setup Iceberg again . or do I leave that branch around and just sync back up to master ? 
btw, I'm guessing that you know I'm Charles not that it matters :)

**Comment by amiracam** [comment]
Surprised not much of a choice there :)  https://github.com/orgs/pharo-project/projects/8.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com">Build software better, together</a></strong></div><div>GitHub is where people build software. More than 40 million people use GitHub to discover, fork, and contribute to over 100 million projects.</div></blockquote>

**Comment by Ducasse** [comment]
Hi Charles

Thanks for your contributions. We will have a look. And I know that feeling!
Fix what you want :). Some of the first issue I did in Pharo was editing the template of the main repo because I was typing it 10 times a day. 
So there is no stupid fixes. Now to relax I'm adding comments and executable comments 
"1 + 2 >>> 3"
The new version of the test runner will check those. 

We created and let some "easy" issues to let people a chance to participate and we can focus on more demanding issues. 

There is no need to delete your fork. 
I can explain to you if need but first can you read the link?
https://github.com/pharo-project/pharo/wiki/Contribute-a-fix-to-Pharo

In a nutshell when you do a Repair from your image state (usually latest) iceberg fetches the commits (between your image state and the repo) and pushes them locally. Then when you do a push to your repo it gets updated (with those changes). 

Stef


<blockquote><div> GitHub</div><div><strong><a href="https://github.com/pharo-project/pharo">pharo-project/pharo</a></strong></div><div>Pharo is a dynamic reflective pure object-oriented language supporting live programming inspired by Smalltalk. - pharo-project/pharo</div></blockquote>

**Comment by amiracam** [comment]
the new version being Dr. Test ? Can a method have multiple comment lines , some executable , some not ?I would assume so. 

Does Travis CI also somehow engage these executable comments ?

 Anyhow the pattern is "%evaluable code%>>>%print of evaluation%" ?
Yes I read that link above , its how I learned how to use Iceberg to fork and contribute in the first place.
So I have done that once obviously and it seems that I should repair to basically pull in the latest and for a subsequent contribution I would create another branch.

BTW I have been using "git-flow" , maybe under the covers Pharo is doing is similar ?
see here: 
https://www.git-tower.com/learn/git/ebook/en/command-line/advanced-topics/git-flow

My day to day life would have been easier and would be easier more relaxing even more fun were I able to code just in Smalltalk. So it would seem in my best interest to help it stay around :)

<blockquote><div><strong><a href="https://www.git-tower.com/learn/git/ebook/en/command-line/advanced-topics/git-flow">Workflows with git-flow</a></strong></div><div>Our learn section helps you get started with various web and software skills. Free online books, videos, and ebooks get you off the ground as quickly as possible.</div></blockquote>

**Comment by Ducasse** [comment]
Check the bug entries list. Because we tag them. And in the project we have to manually add them there. 

You can mix comments and executable comments. 
Now the validator expect that executable comment are only one per comment expression i.e.
"1+2 >>> 3"
"12+0>>>12"
BTW >>> is a message. 

About your last point. We decided to make Pharo to open the space and make sure that people can make a living with it. It was that or going to Python or Lua. So we started and decided that people can join and that we can try to change the world :) and have fun. 
Still Pharo consortium is super concerned into solutions that help people making business. 
For us this is key.

**Comment by amiracam** [comment]
Pharo Consortium is key  for me because for me to be able to sell Pharo as a solution I must show that there's a community around it but also that there's an active "entity" ready to provide some sort of SLA whether needed or not.

**Comment by Ducasse** [comment]
Yes this is really why we created and it took me a lot of time. It took me many discussions and more. But this is nice. This is also why I chased success stories.
Yes we wanted to offer Long term support and we wanted to create PharoPro but we cannot do everything. Sadly but we are working on such offer. 
Did you look at the success stories? 
 This is strange that sometimes people do not understand that we should increase our PR. 
This is also why the consortium created flyers.
